lib/ansible/cli/scripts/ansible_connection_cli_stub.py

```diff
@@ -138,6 +138,10 @@ def run(self):
                     if log_messages:
                         display.display("jsonrpc request: %s" % data, log_only=True)
 
+                    request = json.loads(to_text(data, errors='surrogate_then_replace'))
```

Comment:
Is `surrogate_then_replace`okay here?  It has the potential to mangle data but pass malformed data.

I see that you're only using `request['method'] == 'exec_command`` so maybe it's okay to do that but I thought I should check.  Typically we use `errors=surrogate_or_strict` when dealing with information that will be sent back to the computer.  (And then handle a traceback as malformed data).

Reply:
The text data is the jsoncrpc request string which consists of information related to the method invoked from module side code on the connection object `(eg: Connection(module._socket_path).get())`. As you mentioned in the comment I now think that `surrogate_or_strict` is the right option here.